### DISCUSSION TITLE
Is it possible to download from a local folder

### DISCUSSION THREAD

**Discussion by HerzMichi** [zauberzeug/nicegui#1323] [Q&A] [answered]
### Question

```python
from nicegui import ui

ui.button('NiceGUI Logo', on_click=lambda: ui.download('https://nicegui.io/logo.png'))

ui.run()
``` 

Will download a file from the link. But what if I want to download a file that is locally stored. 
Ty in advance :)

**Comment by falkoschindler** [comment, marked as answer]
Hi @HerzMichi!

For downloading local files, you need to _serve_ first using something like `app.add_static_files`:

```py
from nicegui import app, ui

app.add_static_files('/my_static_files', 'website/static')
ui.button('NiceGUI Logo', on_click=lambda: ui.download('/my_static_files/logo.png'))

ui.run()
```

- "my_static_files" is a route of your choice.
- "website/static" is a folder relative to the current working directory, the root path of the NiceGUI repo in this example.

We are currently working on supporting `ui.download` with local file paths (PR #1118), but we are not finished yet. Hopefully we will be able to simply pass a local file path to `ui.download` and the file is served automatically.

**Reply by HerzMichi** [reply]
Reading your answer and the function definition made it work!
Danke :)
